programs/fundamentals/math.sc

```diff
@@ -42,6 +44,35 @@ distance(vec1, vec2) -> sqrt(reduce(vec1 - vec2, _a + _*_, 0));
 
 dot(vec1, vec2) -> reduce(vec1 * vec2, _a + _, 0);
 
+norm(vec) -> sqrt(reduce(vec, _a + _*_, 0));
+
+normalize(vec) -> vec / sqrt(reduce(vec, _a + _*_, 0));
+
+cross(v, w) -> [v:1*w:2 - v:2*w:1, v:2*w:0 - v:0*w:2, v:0*w:1 - v:1*w:0];
+
+outer(v, w) -> map(w, v*_);
+
+cross_matrix(vec) -> (
+	[x, y, z] = vec;
+	[[0, -z, y], [z, 0, -x], [-y, x, 0]]
+);
+
+// the following methods assume a square matrix indexed by row (i.e, mat:0 is the first row)
+mat_times_vec(matrix, vec) -> map(matrix * map(vec, vec), reduce(_, _a+_, 0));
+
+mat_times_mat(mat1, mat2) -> (
+	mat2 = transpose(mat2);
+	map(mat1,
+		i = _i;
+		map(mat2,
+			j = _i;
+			dot(mat1:i, mat2:j)
+		)
+	)
+);
+
+transpose(mat) -> map(mat, i = _i; map(_, mat:_i:i) );
```

Comment:
Umm, this doesn't work on non-square matrices... `transpose([[1, 2], [3, 4], [5, 6]])` should return `[[1, 3, 5], [2, 4, 6]]`, instead it returns: `[[1, 3], [2, 4], [1, 3]]`

Reply:
True, true. Should add a disclaimer? I'm not gonna fix it, but we can delete it, if you don't like the disclaimer. People are ognna open the app to know what operations are available anyway.

Reply:
I have an idea to how to fix it, but i dont know how to program scarpet, i can do a fix using python structure if it usefull

Reply:
Yeah, post python code here, i'll import it to scarpet.